Code/GraphMol/MonomerMol/MonomerMol.cpp

```diff
@@ -0,0 +1,339 @@
+            continue;
+        }
+        if (get_polymer_id(bond->getBeginAtom()) == polymer_id &&
+            get_polymer_id(bond->getEndAtom()) == polymer_id) {
+            bonds.push_back(bond->getIdx());
+        }
+    }
+
+    std::string annotation{};
+    for (const auto& sg : ::RDKit::getSubstanceGroups(monomer_mol)) {
+        if ((sg.getProp<std::string>("TYPE") != "COP") ||
+            !sg.hasProp(ANNOTATION) || !sg.hasProp("ID")) {
+            continue;
+        }
+        if (sg.getProp<std::string>("ID") == polymer_id) {
+            annotation = sg.getProp<std::string>(ANNOTATION);
+            break;
+        }
+    }
+    return {atoms, bonds, annotation};
+}
+
+ChainType to_chain_type(std::string_view chain_type)
+{
+    if (chain_type == "PEPTIDE") {
+        return ChainType::PEPTIDE;
+    } else if (chain_type == "RNA") {
+        return ChainType::RNA;
+    } else if (chain_type == "DNA") {
+        return ChainType::DNA;
+    } else if (chain_type == "CHEM") {
+        return ChainType::CHEM;
+    } else {
+        throw std::invalid_argument("Invalid chain type");
+    }
+}
+
+std::string to_string(ChainType chain_type)
+{
+    switch (chain_type) {
+        case ChainType::PEPTIDE:
+            return "PEPTIDE";
+        case ChainType::RNA:
+            return "RNA";
+        case ChainType::DNA:
+            return "DNA";
+        case ChainType::CHEM:
+            return "CHEM";
+        default:
+            throw std::invalid_argument("Invalid chain type");
+    }
+}
+
+void add_connection(RDKit::RWMol& monomer_mol, size_t monomer1, size_t monomer2,
+                    const std::string& linkage)
+{
+    const auto new_total =
+        monomer_mol.addBond(monomer1, monomer2, ::RDKit::Bond::BondType::SINGLE);
+    monomer_mol.getBondWithIdx(new_total - 1)->setProp(LINKAGE, linkage);
+    if (linkage == BRANCH_LINKAGE) {
+        // monomer2 is a branch monomer
+        monomer_mol.getAtomWithIdx(monomer2)->setProp(BRANCH_MONOMER, true);
+    }
+}
+
+void add_connection(RDKit::RWMol& monomer_mol, size_t monomer1, size_t monomer2,
+                    ConnectionType connection_type)
+{
+    switch (connection_type) {
+        case ConnectionType::FORWARD:
+            add_connection(monomer_mol, monomer1, monomer2, BACKBONE_LINKAGE);
+            break;
+        case ConnectionType::SIDECHAIN:
+            add_connection(monomer_mol, monomer1, monomer2, BRANCH_LINKAGE);
+            break;
+    }
+}
+
+std::unique_ptr<Monomer> make_monomer(std::string_view name,
+                                      std::string_view chain_id,
+                                      int residue_number, bool is_smiles)
+{
+    auto a = std::make_unique<::RDKit::Atom>();
+    std::string n{name};
+    a->setProp(ATOM_LABEL, n); // temporary, for image generation
+    // Always start with BRANCH_MONOMER as false, will be set to
+    // true if branch linkage is made to this monomer
+    a->setProp(BRANCH_MONOMER, false);
```

Comment:
do we need these?

Reply:
No; I left them to make merging the HELM parser/coordinate generation/drawing code easier in the future, but it seems to just make this more confusing for now so I'll remove it

Reply:
ahh, I see. I feel like without the context of our image gen and HELM stuff it's a bit confusing